src/evaluation_system/tests/plugin_manager_test.py

```diff
@@ -12,13 +13,54 @@
 import time
 from pathlib import Path
 from tempfile import TemporaryDirectory
+from unittest.mock import MagicMock, patch
 
 import pytest
 
-from evaluation_system.api.plugin_manager import PluginMetadata
+from evaluation_system.api.plugin_manager import (
+    CommandConfig,
+    PluginAbstract,
+    PluginMetadata,
+    _PluginStateHandle,
+)
+from evaluation_system.misc.exceptions import ConfigurationException
+from evaluation_system.model.user import User
 
 log = logging.getLogger(__name__)
 
+# def test_update_plugin_state_in_db():
+#     user_mock = MagicMock(spec=User)
+#     plugin_state_handle = _PluginStateHandle(status=1, rowid=123, user=user_mock)
+#     plugin_state_handle._update_plugin_state_in_db()
+#     user_mock.getUserDB.return_value.upgradeStatus.assert_called_once_with(123, 'test_user', 1)
+
+
+@patch("evaluation_system.api.plugin_manager.atexit.register")
+@patch("evaluation_system.api.plugin_manager.signal.signal")
+def test_signal_handling_updates_state_and_quits(mock_signal, mock_atexit_register):
+    user_mock = MagicMock(spec=User)
+    plugin_state_handle = _PluginStateHandle(status=1, rowid=123, user=user_mock)
+    with patch.object(plugin_state_handle, "_update_plugin_state_in_db") as mock_update:
+        exception_caught = False
+        try:
+            plugin_state_handle._update_plugin_state_in_db_and_quit(None, None)
+        except KeyboardInterrupt:
+            exception_caught = True
+        assert exception_caught == True, "KeyboardInterrupt was not raised as expected"
```

Comment:
I do understand what you're doing here but I am not a big fan of it. The whole signal hooking is a) not pretty b) non-trivial and hence c) error prone. 

I think we would be better of, if we did some functional testing here rather than "just" calling the classes and methods involved to expect the obvious. 

I know that it's not pretty to use multiprocessing to submit a plugin, kill it and see if the history entry has changed to 'broken' is in my opinion safer as it simulates what is really happening.

Reply:
Also, looking at my code in the _update_plugin_state_in_db_and_quit method, I think we should not raise a KeyboardInterrupt. I think I did that because somewhere in the code keyboard interrupt are handled. But we could define a custom error class that is also handled. I think a sigint signal is kind of confusing for this. But that's just a general comment (more towards me).

Reply:
yes, it makes more sense to not use the magicmock for unit-testing. To be honest some processes in Classes and funcs are so long and making unit-test for some of them needs a long long unit-test, I though simulating via MagicMock could be the solution to get the full coverage but you are totally right. In the next commit I try to remove more complicated and nested MagicMock-based unit-test. if it is possible please have a look to see which other unit-test needs to be revised.